src/radon.ts

```diff
@@ -1,423 +1,527 @@
 
-  public generateSelectedReducerArgument(reducer: Reducer): MarkupSelectedOption {
-    const selectedArgument: MarkupSelectedOption = {
-      arguments: [],
-      label: Reducer[reducer],
-      hierarchicalType: MarkupHierarchicalType.SelectedOperatorOption,
-      markupType: MarkupType.Option,
-      outputType: OutputType.Bytes,
+  public addOperator() {
+    const lastOutputType = this.getOutputType()
+    const type: Type | null = fromOutputTypeToType(lastOutputType)
+
+    if (type) {
+      const operator: MirOperator = getDefaultMirOperatorByType(type)
+      this.operators.push(
+        new Operator(this.cache, this.scriptId, lastOutputType, operator, this.onChildrenEvent())
+      )
+    } else {
+      // TODO: search in operators the type for the regarding types:
+      // SubscriptOutput, ReducerOutput, FilterOutput, MatchOutput, Same, Inner
+      this.operators.push(
+        new Operator(this.cache, this.scriptId, lastOutputType, null, this.onChildrenEvent())
+      )
     }
-    return selectedArgument
   }
+}
 
-  // TODO: Remove unknown to have a stronger type
-  public unwrapSource(source: CacheRef): MarkupSource {
-    const cachedMarkupSource: CachedMarkupSource = (this.unwrapResultFromCache(
-      source
-    ) as unknown) as CachedMarkupSource
-    const markupSource: MarkupSource = {
-      url: cachedMarkupSource.url,
-      script: this.unwrapScript(cachedMarkupSource.script),
-    }
-
-    return markupSource
+export class Operator {
+  public cache: Cache
+  public operatorInfo: OperatorInfo
+  public code: OperatorCode
+  public mirArguments: MirArgument[] | []
+  public arguments: Array<Argument>
+  public default: Boolean
+  public scriptId: number
+  public inputType: OutputType
+  public id: number
+  public eventEmitter: EventEmitter
+
+  constructor(
+    cache: Cache,
+    scriptId: number,
+    inputType: OutputType | null,
+    operator: MirOperator | null,
+    eventEmitter: EventEmitter
+  ) {
+    const { code, args } = getMirOperatorInfo(operator || DEFAULT_OPERATOR)
+    this.eventEmitter = eventEmitter
+    this.id = cache.insert(this).id
+    this.default = !operator
+    this.cache = cache
+    this.code = code
+    this.operatorInfo = operatorInfos[code]
+    this.mirArguments = args
+    this.inputType = inputType || DEFAULT_INPUT_TYPE
+    this.arguments = args.map(
+      (x, index: number) => new Argument(cache, this.operatorInfo.arguments[index], x)
+    )
+    this.scriptId = scriptId
   }
-  public unwrapScript(script: Array<CacheRef>): MarkupScript {
-    const markupScript: MarkupScript = script.map(operatorRef => {
-      const cachedOperator: CachedMarkupOperator = (this.unwrapResultFromCache(
-        operatorRef
-      ) as unknown) as CachedMarkupOperator
-      const operator: MarkupOperator = this.unwrapOperator(cachedOperator, operatorRef.id)
-
-      return operator
-    })
 
-    return markupScript
+  public update(args: { label?: OperatorName; code?: OperatorCode }) {
```

Comment:
What's the problem with:

```suggestion
  public update(args: OperatorName | OperatorCode ) {
```

Reply:
This is an straightforward way to avoid a match between OperatorName and OperatorCode